public/app/features/dashboard/components/PanelEditor/PanelOptionsTab.tsx

```diff
@@ -41,47 +44,87 @@ export const PanelOptionsTab: FC<Props> = ({
 
   const maxPerRowOptions = [2, 3, 4, 6, 8, 12].map(value => ({ label: value.toString(), value }));
 
-  const focusVisPickerInput = (isExpanded: boolean) => {
+  const focusVisPickerInput = (onSaveToStore: StoreOptionGroupCallback) => (isExpanded: boolean) => {
     if (isExpanded && visTabInputRef.current) {
       visTabInputRef.current.focus();
     }
+    storeOptionGroupExpanded(isExpanded, onSaveToStore);
   };
   // Fist common panel settings Title, description
   elements.push(
-    <OptionsGroup title="Panel settings" key="Panel settings">
-      <Field label="Panel title">
-        <Input defaultValue={panel.title} onBlur={e => onPanelConfigChange('title', e.currentTarget.value)} />
-      </Field>
-      <Field label="Description" description="Panel description supports markdown and links.">
-        <TextArea
-          defaultValue={panel.description}
-          onBlur={e => onPanelConfigChange('description', e.currentTarget.value)}
-        />
-      </Field>
-      <Field label="Transparent" description="Display panel without a background.">
-        <Switch value={panel.transparent} onChange={e => onPanelConfigChange('transparent', e.currentTarget.checked)} />
-      </Field>
-    </OptionsGroup>
+    <LocalStorageValueProvider<OptionGroupStorageItem>
+      storageKey={getOptionGroupStorageKey('Panel settings')}
+      defaultValue={defaultStorageItem}
+      key="Panel settings"
+    >
+      {(values, onSaveToStore) => (
+        <OptionsGroup
+          title="Panel settings"
+          defaultToClosed={values.defaultToClosed}
+          onToggle={onOptionGroupToggle(onSaveToStore)}
+        >
```

Comment:
Since we use OptionsGroups in so many places this will be duplicated in many places. So having this be internal to the OptionsGroup component looks preferable, what was the drawback to that approach?

Reply:
We need to add `id` prop for instance, and OptionGroup will be tightly coupled with local storage as `storage` is 2 on the top of my mind. But if you feel thats a better approach I'm all for it

Reply:
Hmmm, I was thinkng more about LocalStorageValueProvider being an input for a `CollapsibleSectionPersistanceContext` (or something like this) that the OptionsGroup could use. Then we could control via prop whether or not a group state should be persisted:
```
<OptionsGroup
  rememberState 
  ... 
/>
```

Reply:
sure, I'll have another PR up soon!

Reply:
Think we gonna need OptionGroup to implement `CollapsibleSection` or `CollapsibleSectionWithPersistence` depending on the prop being passed or not. Then we could still test the CollapsibleSection independently from local storage.